File: application/backend-credit-card-enrollment/backend-java/src/main/java/cloud/ambar/creditcard/enrollment/controller/EnrollmentCommandController.java
package cloud.ambar.creditcard.enrollment.controller;

import cloud.ambar.common.commandhandler.CommandController;
import cloud.ambar.common.eventstore.EventStore;
import cloud.ambar.common.sessionauth.SessionService;
import cloud.ambar.creditcard.enrollment.commandhandler.RequestEnrollmentCommandHandler;
import cloud.ambar.creditcard.enrollment.commandhandler.RequestEnrollmentCommand;
import jakarta.validation.Valid;
import org.springframework.http.HttpStatus;
import org.springframework.web.bind.annotation.*;
import org.springframework.web.context.annotation.RequestScope;

@RestController
@RequestScope
@RequestMapping("/api/v1/credit_card/enrollment")
public class EnrollmentCommandController extends CommandController {
    private final SessionService sessionService;

    private final RequestEnrollmentCommandHandler requestEnrollmentCommandHandler;

    public EnrollmentCommandController(
            EventStore eventStore,
            SessionService sessionService,
            RequestEnrollmentCommandHandler requestEnrollmentCommandHandler
    ) {
        super(eventStore);
        this.sessionService = sessionService;
        this.requestEnrollmentCommandHandler = requestEnrollmentCommandHandler;
    }

    @PostMapping("/request-enrollment")
    @ResponseStatus(HttpStatus.OK)
    public void requestEnrollment(
            @RequestHeader("X-With-Session-Token") String sessionToken,
            @Valid @RequestBody RequestEnrollmentHttpRequest request) {
        final RequestEnrollmentCommand command = RequestEnrollmentCommand
                .builder()
                .userId(sessionService.authenticatedUserIdFromSessionToken(sessionToken))
                .productId(request.getProductId())
                .annualIncomeInCents(request.getAnnualIncomeInCents())
                .build();

        processCommand(command, requestEnrollmentCommandHandler);
    }
}

File: application/backend-credit-card-enrollment/backend-c#/CreditCard/Enrollment/Command/EnrollmentCommandController.cs
using CreditCardEnrollment.Common.Command;
using CreditCardEnrollment.Common.EventStore;
using CreditCardEnrollment.Common.Projection;
using Microsoft.AspNetCore.Mvc;

namespace CreditCardEnrollment.CreditCard.Enrollment.Command;

[ApiController]
[Route("api/v1/credit_card/enrollment")]
[Produces("application/json")]
[Consumes("application/json")]
public class EnrollmentCommandController : CommandController {
    private readonly RequestEnrollmentCommandHandler _requestEnrollmentCommandHandler;

    public EnrollmentCommandController(
        PostgresTransactionalEventStore postgresTransactionalEventStore,
        MongoTransactionalProjectionOperator mongoTransactionalProjectionOperator,
        ILogger<EnrollmentCommandController> logger,
        RequestEnrollmentCommandHandler requestEnrollmentCommandHandler) 
        : base(postgresTransactionalEventStore, mongoTransactionalProjectionOperator, logger) {
        _requestEnrollmentCommandHandler = requestEnrollmentCommandHandler;
    }

    [HttpPost("request-enrollment")]
    [ProducesResponseType(typeof(object), StatusCodes.Status200OK)]
    public IActionResult RequestEnrollment(
        [FromHeader(Name = "X-With-Session-Token")] string sessionToken,
        [FromBody] RequestEnrollmentHttpRequest request) {
        var command = new RequestEnrollmentCommand {
            SessionToken = sessionToken,
            ProductId = request.ProductId,
            AnnualIncomeInCents = request.AnnualIncomeInCents
        };

        ProcessCommand(command, _requestEnrollmentCommandHandler);
        return new OkObjectResult(new {});
    }
}